### DISCUSSION TITLE
Secure hadoop integration with hue as proxyuser

### DISCUSSION THREAD

**Discussion by SoniaComp** [cloudera/hue#3323] [Questions] [answered]
### Description

Hello!

I am working on integrating Hue with SecureHadoop as a ProxyUser.
I found that the behavior of ProxyUser in WebHDFS FileBrowser needs to be enhanced.
When "security_enabled" is true, [only "read_url" method use delegation token](https://github.com/cloudera/hue/blob/master/desktop/libs/hadoop/src/hadoop/fs/webhdfs.py#L565).

I want to modify the code 
so that a user who has been impersonated from Hue(a proxy user) can issue a Delegation Token when using WebHDFS.

**Comment by bjornalm** [comment, marked as answer]
Hi @SoniaComp, thanks for reaching out. :-) 
Since Hue is open source you can clone the Hue repo and modify the code. If you want to share your changes and get them merged into the next Hue release you can create a PR with your changes.  

See 
https://github.com/cloudera/hue/blob/master/CONTRIBUTING
https://docs.gethue.com/developer/development/
